src/Control/Distributed/Process/Node.hs

```diff
@@ -389,7 +390,9 @@ forkProcess node proc =
                 (return . DiedException . (show :: SomeException -> String)))]
 
           -- [Unified: Table 4, rules termination and exiting]
-          modifyValidLocalState_ node (cleanupProcess pid)
+          mconns <- modifyValidLocalState node (cleanupProcess pid)
+          forM_ mconns (mapM_ NT.close)
```

Comment:
You're saying `forkIO` [here](https://github.com/haskell-distributed/distributed-process/pull/249/files#diff-feba37ab068c8749d02e4a8c30d18d41R163), whereas you're not here. Why is that? I would argue the NT.close should always be one-sided and asynchronous, hence no need to `forkIO`. But, the current API makes it difficult indeed for transports to guarantee that, so maybe `forkIO` consistently here but add an XXX with a reference to the ticket about making the network-transport API asynchronous?